server/bin/pbench-audit-server.py

```diff
@@ -0,0 +1,581 @@
+
+    try:
+        config = PbenchServerConfig(cfg_name)
+    except BadConfig as e:
+        print(f"{_NAME_}: {e} (config file {cfg_name})", file=sys.stderr)
+        return 1
+
+    if check_directory_exists(config.ARCHIVE, "ARCHIVE") > 0:
+        return 1
+
+    if check_directory_exists(config.INCOMING, "INCOMING") > 0:
+        return 1
+
+    if check_directory_exists(config.RESULTS, "RESULTS") > 0:
+        return 1
+
+    if check_directory_exists(config.USERS, "USERS") > 0:
+        return 1
+
+    logger = get_pbench_logger(_NAME_, config)
+    init_db(config, logger)
+
+    ret = 0
+
+    try:
+        Path(config.LOGSDIR, "pbench-audit-server").mkdir()
+    except FileExistsError:
+        # directory already exists, ignore
+        pass
+    except Exception:
+        print("os.mkdir: Unable to create destination directory")
+        return 1
+    logfile = Path(config.LOGSDIR, "pbench-audit-server", "pbench-audit-server.log")
+    # TEMPORARY addition of error file for the sake of test cases
+    errorfile = Path(config.LOGSDIR, "pbench-audit-server", "pbench-audit-server.error")
+    with errorfile.open(mode="w") as f:
+        pass
+    # END
+
+    logger.info("start-{}", config.TS)
+
+    with logfile.open(mode="w") as f:
+
+        ahier = ArchiveHierarchy("archive", config.ARCHIVE, config)
+        verify_archive(ahier)
+        cnt = 0
+        if ahier.check_controller():
+            f.write(f"\nstart-{config.TS[4:]}: archive hierarchy: {config.ARCHIVE}\n")
+            cnt = ahier.dump(f)
+            ahier.header(f, "end")
+        if cnt > 0:
+            ret += 1
+
+        ihier = IncomingHierarchy("Incoming", config.INCOMING, config)
+        cihier = ControllerHierarchy("incoming", config.INCOMING, config)
+        verify_controllers(cihier)
+        verify_incoming(ihier, cihier.verifylist)
+        cnt = check_and_dump(f, cihier, ihier)
+        if cnt > 0:
+            ret += 1
+
+        rhier = ResultsHierarchy("Results", config.RESULTS, config)
+        crhier = ControllerHierarchy("results", config.RESULTS, config)
+        verify_controllers(crhier)
+        verify_results(rhier, crhier.verifylist)
+        cnt = check_and_dump(f, crhier, rhier)
+        if cnt > 0:
+            ret += 1
+
+        uhier = UserHierarchy("users", config.USERS, config)
+        verify_users(uhier)
+        if uhier.users:
+            for user in uhier.users:
+                verify_results(user, crhier.verifylist)
+        cnt = 0
+        if uhier.check_controller():
+            uhier.header(f, "start")
+            cnt = uhier.dump(f)
+            uhier.header(f, "end")
+        if cnt > 0:
+            ret += 1
+
+    # prepare and send report
+    with tempfile.NamedTemporaryFile(mode="w+t", dir=config.TMP) as reportfp:
+        with open(logfile, "r") as f:
+            reportfp.write(
+                f"{_NAME_}.run-{config.timestamp()}({config.PBENCH_ENV})\n{f.read()}"
+            )
+            reportfp.seek(0)
```

Comment:
They are accessed at the time of creating reports

Reply:
I'm not sure what you mean.  I see the report opened at line 562, written at line 564, rewound at line 567, and then the file's _name_ is referenced in the call at line 572, but the file object `reportfd` is never referenced again.  Deep inside `post_status()`, it opens the file again, at which point it will start reading from the beginning, regardless of what we do in this function.

So, I think you can omit the `seek()` call.

Reply:
I tried removing it but it is throwing an error "Report post unsuccessful: Logic Bomb!" in the reports.py file.

Reply:
I'm going to start a new "conversation" for this, based on the current code.

Reply:
[I changed my mind....]

The problem here is that the code is calling `post_status()` and passing the report file (by name) when the calling code still has the file open!  The `seek()` presumably has a side-effect of flushing the writes made so far to disk, such that, without it, `post_status()` sees an empty file.  The correct thing to do is to close the file before handing it off to `post_status()`.

So, the code after line 557 should be out-dented (moved out of the `with` block), so that it runs after the file is closed. (And, then you should be able to remove the call to `seek()` without things breaking.)